File: app/src/main/java/com/easy/kotlins/http/core/extension/OkExtension.java
package com.easy.kotlins.http.core.extension;


import com.easy.kotlins.http.core.OkFaker;

/**
 * Create by LiZhanPing on 2020/8/26
 */
public abstract class OkExtension {

    private OkFaker mOkFaker;


    public OkFaker getOkFaker() {
        if(mOkFaker == null){
            throw new NullPointerException("you must install extension to OkFaker");
        }
        return mOkFaker;
    }


    public final void install(OkFaker faker) {
        mOkFaker = faker;
    }

}

File: common-ktx/src/main/java/com/easy/kotlins/http/core/extension/OkExtension.kt
package com.easy.kotlins.http.core.extension

import com.easy.kotlins.http.core.OkRequest
import okhttp3.Request
import okhttp3.Response

/**
 * Create by LiZhanPing on 2020/8/26
 */
abstract class OkExtension {
    private var _request: OkRequest? = null

    val request: OkRequest
        get() {
            return _request ?: throw NullPointerException("you must install extension to OkFaker")
        }

    fun install(faker: OkRequest?) {
        _request = faker
    }

    abstract fun onRequest(builder: Request.Builder)

    abstract fun onResponse(response: Response): Boolean

    abstract fun onError(error: Throwable): Boolean
}